pkg/operator/ceph/object/rgw-probe.sh

```diff
@@ -17,11 +17,12 @@ RGW_URL="$PROBE_PROTOCOL://0.0.0.0:$PROBE_PORT"
 
 function check() {
   local URL="$1"
+  # --max-time 3 - override the default 2 minutes wait time down to 3 seconds
   # --insecure - don't validate ssl if using secure port only
   # --silent - don't output progress info
   # --output /dev/stderr - output HTML header to stdout (good for debugging)
   # --write-out '%{response_code}' - print the HTTP response code to stdout
-  curl --insecure --silent --output /dev/stderr --write-out '%{response_code}' "$URL"
+  curl --max-time 3 --insecure --silent --output /dev/stderr --write-out '%{response_code}' "$URL"
```

Comment:
I would prefer to parameterize the `--max-time` value instead of hard-coding it. When we have had hard-coded things like this in previous probes, it has led to situations where some users are encountering an error state that they are unable to work around.

Let's add a `PROBE_TIMEOUT="{{ .Timeout }}` value to the beginning of this script and use `--max-time "$PROBE_TIMEOUT"`. Because the bash portion of the script should process in under a second, I'm not worried about a few hundred milliseconds of time where something extra may be running.

This will require passing the configured probe timeout to the template config in .go code.

Reply:
Don't forget to update the commit title and body, and the PR description to reflect that the probe is no longer hard-coded to 3 seconds.